### ISSUE TITLE
ASP.net: System.AccessViolationException

### ISSUE THREAD

**Issue by boeledi** [dlemstra/Magick.NET#76]
Hi,

I obtain the following exception when using inside an ASP.NET (MVC 5, .NET 4.5.2) web application:

System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'

Here is the source code:

MagickReadSettings settings = new MagickReadSettings();
settings.Density = new Density(300);

                using (MagickImage image = new MagickImage(pngFilename, settings))
                {
                    MagickGeometry size = new MagickGeometry(_width, _height);
                    size.IgnoreAspectRatio = true;
                    image.Resize(size);
                    image.Write(thumbnailFilename);
                }

The error occurs at the image.Resize(size) level.

Version: Margick.NET-Q16-AnyCPU 7.0.6.102

Regards,

**Comment by dlemstra** [comment]
Can you share your input image and the values you use for _width and _height?

**Comment by dlemstra** [comment]
Is your issue resolved @boeledi?

**Comment by pcoffey42** [comment]
I don't know if  @boeledi still has the issue, but I've just recently been experiencing it as well.

Here's is my code, and the attached pdf that caused the error

                            using (var fileStream = new FileStream(pdfFilePath, FileMode.Open))
                            {
                                MagickReadSettings settings = new MagickReadSettings();
                                // Settings the density to 300 dpi will create an image with a better quality
                                settings.Density = new Density(600, 600);

                                using (MagickImageCollection images = new MagickImageCollection())
                                {
                                    // Add all the pages of the pdf file to the collection
                                    images.Read(fileStream, settings);

                                    int page = 1;
                                    foreach (MagickImage image in images)
                                    {

                                        using (var pictureAsStream = new System.IO.MemoryStream())
                                        {



                                            image.Write(pictureAsStream);
                                        }
                                    }
                                }
                            }
[errorpdf.pdf](https://github.com/dlemstra/Magick.NET/files/1198386/errorpdf.pdf)

**Comment by dlemstra** [comment]
I am unable to reproduce the issue on my machine. Is it possible to create a crashdump or get some more information from the eventviewer?

**Comment by pcoffey42** [comment]
It seems to only be happening on a seperate thread and doesn't offer me the chance to pull in a crash dump. I'll see if there's another way for me to get it and post here.

I'm using the Q8-AnyCpu version of the app. Is it possible I'm using the wrong version? (Probably a stupid question, it was just a random thought)

**Comment by dlemstra** [comment]
There is no wrong version 😃  Are you in a x64 or a x86 application pool? And did you install the x64 or the x86 version of Ghostscript?

**Comment by pcoffey42** [comment]
It seems to only happen when the pdf has more than one image.
I'm going through now to check to make sure the correct version of everything is installed. I'll let you know if that's what caused my issue.

**Comment by dlemstra** [comment]
@pcoffey42 Any news?

**Comment by louiscloutierdakis** [comment]
@dlemstra I have the same error System.AccessViolationException for resize. I have an intel hd 2500 display adapter with windows 7.
I used Magick.NET-Q16-AnyCPU 7.0.7

Just doing something like this with any image and any sizes cause this error on the resize line.
```
using (MagickImage image = new MagickImage(photo.OrderFilePath))
{
    image.Resize(200, 200);
```

**Comment by louiscloutierdakis** [comment]
@dlemstra just for information, I fixed my problem by updating the driver to the latest one. Now everything is working.

**Comment by dlemstra** [comment]
If it happened inside the resize method is was probably another issue. It was most likely an OpenCL issue and not related to the issue @pcoffey42 is experiencing.

**Comment by ljani** [comment]
I actually started seeing this error *after* updating my drivers to the latest versions for some other reasons. It was working fine before.

My laptop has this Nvidia Optimus technology meaning it has both, an Intel HD Graphics 530 and Nvidia GeForce GTX 960M graphics cards. The newest version of the Intel drivers as of today is 15.46.05.4771 dated 2017-08-23 and the Nvidia driver version is 385.69 released on 2017-09-21.

Optimus provides me an option to configure the graphics card used per process, but it doesn't seem to have any effect on this. If I had to guess, I'd blame the Intel's drivers as I've seen similar issues with a different model and much older driver version back in the day. I'd be interested what version of the Intel drivers are you guys using?

One option would be to downgrade the drivers, but since I only need this piece of software once in a blue moon, turning of OpenCL ImageMagick/Magick.NET seems more attracting. To do this, add an environment variable called `MAGICK_OCL_DEVICE` with value `OFF`. This also fixes the issue for me.
